### PR TITLE
Major improvements to electron wrapper

### PR DESCRIPTION
**Pull request by amitojsingh366**
- Fixed issue #1285 
- Temp fix but not really for  issue #1229 
- Added overlay *(windows and linux only, not tested on linux)*
     - Overlay can be used on any non-full screen application
     - Shows the current speaking state of the speakers in the room
     - Custom keybinds for overlay can be set
     - Is very jank

### CODE REVIEW

**Comment by vi241** [comment]
Which language is et?

**Comment by amitojsingh366** [comment]
> 
> 
> Which language is et?

Pardon me i didn't get you?

**Comment by simonri** [comment]
-57k lines moment

**Comment by vi241** [comment]
nvm

**Comment by amitojsingh366** [comment]
> 
> 
> nvm

👍

**Comment by HarrisonMayotte** [comment]
The bulk of the negative lines appear to be from package-lock.json so I'm not concerned about it.

**Review by HarrisonMayotte — APPROVED** [review]
Looks okay to me but should be reviewed by at least one other person.

**Review by HarrisonMayotte — APPROVED** [review]

**Comment by amitojsingh366** [comment]
No way vercel bot actually passed the check

**Comment by HarrisonMayotte** [comment]
It appears that it has.

**Review by HarrisonMayotte — APPROVED** [review]

File: baklava/resources/overlay/asset-manifest.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "files": {
+    "main.css": "./static/css/main.f434b18b.chunk.css",
+    "main.js": "./static/js/main.bc92b591.chunk.js",
+    "main.js.map": "./static/js/main.bc92b591.chunk.js.map",
+    "runtime-main.js": "./static/js/runtime-main.7dea6612.js",
+    "runtime-main.js.map": "./static/js/runtime-main.7dea6612.js.map",
+    "static/js/2.b319a4ae.chunk.js": "./static/js/2.b319a4ae.chunk.js",
+    "static/js/2.b319a4ae.chunk.js.map": "./static/js/2.b319a4ae.chunk.js.map",
+    "static/js/3.5b106a42.chunk.js": "./static/js/3.5b106a42.chunk.js",
```
Comment (benawad) [inline baklava/resources/overlay/asset-manifest.json]: don't check in minified js, we can add a build step instead

**Review by benawad — CHANGES_REQUESTED** [review]

File: baklava/resources/overlay/asset-manifest.json (hunk 1 above)
Comment (amitojsingh366) [reply baklava/resources/overlay/asset-manifest.json]: Didn't get you? Do you want me to shift the react project for the overlay into baklava folder itself?

Comment (amitojsingh366) [reply baklava/resources/overlay/asset-manifest.json]: If yes then wont that increase the size of baklava like a lot? Cuz the node_modules and stuff?

File: baklava/README.md [hunk 2]
```diff
@@ -34,10 +34,11 @@ If you encounter any other errors make a new issue.
 
 > **NOTE:** I have no idea why iohook requires this but it does. And I only found iohook to be a reliable module to detect the `keyup` and `keydown` events *(when app in background)* for use with PTT.
 
-## Using hot reload
+~~## Using hot reload~~
 
-The electron wrapper has the capability for hot reloads during development.
-You can run the app in hot reload mode using `npm run dev`.
+~~The electron wrapper has the capability for hot reloads during development.
+You can run the app in hot reload mode using `npm run dev`.~~
```
Comment (theotterlord) [inline baklava/README.md]: Hot reload is available, but your fork seems to have an incorrect package.json

File: baklava/package.json [hunk 3]
```diff
@@ -1,98 +1,91 @@
 {
-  "name": "dogehouse",
-  "version": "1.0.16",
-  "description": "Taking voice conversations to the moon 🚀",
-  "main": "./dist/electron.js",
-  "scripts": {
-    "compile": "tsc",
-    "watch": "tsc -w",
-    "dev": "npm run compile && node ./dist/dev.js",
-    "start": "npm run compile && electron ./dist/electron.js",
```
Comment (theotterlord) [inline baklava/package.json:10]: This is the dev script. Lots of PRs to baklava seem to remove it for some reason

File: baklava/package.json [hunk 4]
```diff
@@ -1,98 +1,91 @@
...
+    "name": "dogehouse",
+    "version": "1.0.17",
+    "description": "Taking voice conversations to the moon 🚀",
+    "main": "./dist/electron.js",
+    "scripts": {
+        "compile": "tsc",
+        "watch": "tsc -w",
+        "dev": "npm run compile && electron ./dist/electron.js",
+        "build": "npm run compile && electron-builder",
+        "build:all": "npm run compile && electron-builder -mwl --win nsis",
```
Comment (theotterlord) [inline baklava/package.json]: The `nsis` and `--win` tag is not required as nsis is our only build target for windows

**Review by TheOtterlord — CHANGES_REQUESTED** [review]

File: baklava/README.md (hunk 2 above)
Comment (amitojsingh366) [reply baklava/README.md]: Oh my bad. Will fix it

File: baklava/package.json (hunk 3 above)
Comment (amitojsingh366) [reply baklava/package.json:10]: My bad...

File: baklava/package.json (hunk 4 above)
Comment (amitojsingh366) [reply baklava/package.json]: Sure!

**Comment by amitojsingh366** [comment]
@TheOtterlord can you check now if it's all fixed?
also @benawad i added the react app for the overlay in baklava. `baklava/resources/overlay`

**Comment by amitojsingh366** [comment]
ok i knew it was going to be a big PR but didn't know its going to be THIS BIG

**Review by TheOtterlord — APPROVED** [review]
Looks good

**Comment by amitojsingh366** [comment]
> 
> 
> Looks good

👍

File: baklava/package.json (hunk 3 above)
Comment (amitojsingh366) [reply baklava/package.json:10]: Yea idk why this happened tho. I freshly pulled from staging right before submitting this PR but i got the outdated package.json

Comment (theotterlord) [reply baklava/package.json:10]: It's strange. I'm going to look into it sometime this week to see if something is of with the commit history.

**Pull request merged**